### ISSUE TITLE
Allow defining instance type ID for ovirt_vm

### ISSUE THREAD

**Issue by janosdebugs** [haveyoudebuggedit/terraform-provider-ovirt#13]
Allow defining the instance_type_id for the ovirt_vm.

**Comment by ahadas** [comment]
Instance types are deprecated in RHV, we should rather drop existing references to instance types than adding new ones

**Comment by janosdebugs** [comment]
@ahadas we need to support older oVirt version and the OpenShift installer has support for it so we can't drop it.

**Comment by ahadas** [comment]
supporting older ovirt versions is irrelevant here - you can always specify what you need instead of relying on what the instance type `Large` is set with. it's a pity if you've added it also to other components.. I guess you can always keep getting it in the API and instead specify the properties that this instance type comes with by default. anyway, let's not expose it further - it's a deprecated feature that would have been removed already if it was easy to remove it

**Comment by janosdebugs** [comment]
@ahadas not really, the installconfig structure is also not easy to break in terms of backwards compatibility and customers expect to be able to port their installconfig structure forward. That means that the installer needs the ability to pass the instance type ID to oVirt, which is does via Terraform. Removing it is currently not an option.

**Comment by ahadas** [comment]
that's unfortunate that you can't drop it now.. but on the good side, it's unlikely that we'll drop it at this point (but new features/changes may not work - for example, we may not change instance types to use VGA and so VMs that remain with QXL won't start on CentOS 9, that's something you should take into account if you keep using a deprecated feature..

**Comment by ahadas** [comment]
I trust you to sort that out, just please don't add more stuff that relies on that deprecated feature

**Comment by michalskrivanek** [comment]
we should deprecate taht in OCP too then. It may really break, on oVirt side it's deprecated since ages now. It was a mistake it was introduced in OCP because at that time it was already deprecated. 
It's ok when it worked, but we should mark it deprecated here as well so we can drop it eventually.
As for b/w compatibility, I would't bother too much - if/when it breaks in oVirt 4.5 keeping it in OCP won't help you. sooner it disappears the better...
